repair/task_manager_module.hh

```diff
@@ -100,6 +100,33 @@ protected:
     virtual std::optional<double> expected_children_number() const override;
 };
 
+class tablet_repair_task_impl : public repair_task_impl {
+private:
+    std::vector<table_id> _table_ids;
+    dht::token_range_vector _ranges;
+    std::unordered_map<dht::token_range, repair_neighbors> _neighbors;
+    locator::effective_replication_map_ptr _erm;
```

Comment:
If we hold this, it should block tablet migrations, no?

This is bad for the long term, but good for the short term. It ensures correctness over performance.

@tgrabiec

Reply:
@tgrabiec provided an example[1], in tablets design doc, on why repair and migration are mutually exclusive. and then he proceed to suggest how to make the locking fine grained, through coordination between balancer and repair scheduler.

[1]
```
Tablet has replicas on nodes A, B C
Tablet movement is initiated with replica change C -> D
Repair is run on node D
Replica movement completes
Repair is run on node C
Repair is run on the remaining replicas

After this, the whole system is supposedly repaired, but the tablet's replica C is in fact not touched by repair.
```

Reply:
Tablets should use tombstone_gc_mode = repair, and then if we miss a tablet nothing bad happens.

Reply:
How can we do that? Is there any sample code to achieve that?

Reply:
I will provide the code to do that. You can ignore it for now.

Reply:
> Tablets should use tombstone_gc_mode = repair, and then if we miss a tablet nothing bad happens.

I opened https://github.com/scylladb/scylladb/issues/16627 for this.

Reply:
If the last repair time stored locally for tombstone_gc_mode = repair? If so it will quickly get out of date. We should move it to the tablets table.

Reply:
Yes, it is stored in a local table. Will it introduce any circular dependency?  The table needs repair.  The repair needs the table to work.

Reply:
The tablets table is managed by Raft and doesn't need repair.

Reply:
(one day Raft will use repair internally to transfer state, so we'll have to figure something out, but we don't need to worry about it now)

Reply:
> Is the last repair time stored locally for tombstone_gc_mode = repair? If so it will quickly get out of date. We should move it to the tablets table.

You mean it will get out of date because tablet replicas are migrated but repair metadata is not? Moving metadata to tablets table is best, but we can also migrate the metadata as part of tablet migration to avoid churn.